.github/workflows/lint.yml

```diff
@@ -97,18 +97,18 @@ jobs:
           checkout_path: ${{ github.workspace }}
       - name: Check auto-gen files are up-to-date
         run: |
-          echo -e "\n::group:: ===> check auto-gen files are up-to-date..."
-          
+          echo -e "\n::group:: ===> Check auto-gen files are up-to-date..."
           ONNX_ML=1 python onnx/defs/gen_doc.py
           ONNX_ML=0 python onnx/defs/gen_doc.py
           python onnx/gen_proto.py -l
           python onnx/gen_proto.py -l --ml
           python onnx/backend/test/stat_coverage.py
+          echo -e "::endgroup::"
 
+          echo -e "\n::group:: ===> Auto-gen results"
           git status
-          git diff --exit-code -- . ':(exclude)onnx/onnx-data.proto' ':(exclude)onnx/onnx-data.proto3'
-          if [ $? -ne 0 ]; then
-            echo "git diff returned failures"
+          if ! git diff --color --exit-code -- . ':(exclude)onnx/onnx-data.proto' ':(exclude)onnx/onnx-data.proto3'; then
+            echo -e "\e[1m\e[36mGit diff returned failures. This may mean you need to regenerate the docs or need to update the source code that will generate the documentation.\e[0m"
```

Comment:
Maybe we can add "Please see onnx/tools/update_doc.bat/sh for more details." ? Hopefully those files are still accurate.

Reply:
I'm not sure these scripts are working (`update_doc.sh`), they run `python setup.py develop` which is not even used in CI (`pip install -e .`). Maybe it would be worth to switch to a fully automated file. Fixing a typo takes 5 minutes + 5 minutes to recompile everything if you are lucky. But users may have an issue with missing dependencies, a wrong version of protobuf installed, a missing compiler (on Windows for example), ... It seems too much to change one character like in this PR #5351.